server/repository/src/db_diesel/master_list.rs

```diff
@@ -59,14 +59,46 @@ impl<'a> MasterListRepository<'a> {
         self.query(Pagination::new(), Some(filter), None)
     }
 
+    pub fn create_filtered_query(filter: Option<MasterListFilter>) -> BoxedMasterListQuery {
+        let mut query = master_list_dsl::master_list.into_boxed();
+
+        if let Some(f) = filter {
+            apply_equal_filter!(query, f.id, master_list_dsl::id);
+            apply_simple_string_filter!(query, f.name, master_list_dsl::name);
+            apply_simple_string_filter!(query, f.code, master_list_dsl::code);
+            apply_simple_string_filter!(query, f.description, master_list_dsl::description);
+
+            // Result master list should be unique, which would need extra logic if we were to join
+            // name table through master_list_name_join, thus query separately and use resulting
```

Comment:
be025dfabe82d250a30087304f77768d3d558c2f

Reply:
> Haven't tested this but I'm pretty sure this is a sub query

Yep I mentioned in PR description: `(instead of querying ids first and using them in equal_any, do inner query)`

Thanks for catching the comment change